sycl/test-e2e/bindless_images/read_sampled.cpp

```diff
@@ -61,24 +61,23 @@ static bool isNumberWithinPercentOfNumber(float firstN, float percent,
 
 // Return fractional part of argument
 // Whole part is returned through wholeComp
-static double fract(double x, double *wholeComp) {
+static float fract(float x, float *wholeComp) {
   // This fmin operation is to prevent fract from returning 1.0.
   // Instead will return the largest possible floating-point number less
   // than 1.0
-  double fractComp = std::fmin(x - std::floor(x), 0x1.fffffep-1f);
+  float fractComp = std::fmin(x - std::floor(x), 0x1.fffffep-1f);
   *wholeComp = std::floor(x);
   return fractComp;
 }
 
 // Returns the two pixels to access plus the weight each of them have
-static double getCommonLinearFractAndCoordsfp64(double coord, int *x0,
-                                                int *x1) {
-  double pixelCoord;
+static float getCommonLinearFractAndCoordsfp64(float coord, int *x0, int *x1) {
+  float pixelCoord;
 
   // Subtract to align so that pixel center is 0.5 away from origin.
   coord = coord - 0.5;
 
-  double weight = fract(coord, &pixelCoord);
+  float weight = fract(coord, &pixelCoord);
   *x0 = static_cast<int>(std::floor(pixelCoord));
   *x1 = *x0 + 1;
   return weight;
```

Comment:
Although this function name doesn't exactly follow a format, I believe this warrants renaming it to `getCommonLinearFractAndCoordsfp` or even `getCommonLinearFractAndCoords`.

Reply:
done